### PR TITLE
pipe shell to pass java port when constructing an arc

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: shells/pipes-shell/device.js [hunk 1]
```diff
@@ -72,6 +72,20 @@ const shellApi = {
     log('flushing caches');
     Utils.env.loader.flushCaches();
     marshalRecipeContext();
+  },
+  registerMessageHandler(handler) {
```
Comment (shaper) [inline shells/pipes-shell/device.js]: We've shifted Android to build on pipes-shell-2, should `pipes-shell` be deprecated and we should make the needed changes in this PR there instead? Or is `pipes-shell` still a thing and we need to evolve it? @yuangu-google

**Comment by shaper** [comment]
Two other notes re: my other comment:
1. I am guessing perhaps this is modifying pipes-shell because javaharness which we're using for JavaPEC work is still using pipes-shell not pipes-shell-2, can we look at moving it to use pipes-shell-2 as a separate PR before this one?
2. Worth adding note to the PR description on what the function need is that prompted this PR

File: shells/pipes-shell/device.js (hunk 1 above)
Comment (yuangu-google) [reply shells/pipes-shell/device.js]: We don't use pipes-shell in Android any more. Only pipes-shell-2 is used.

**Comment by mmandlis** [comment]
Closing this PR - https://github.com/PolymerLabs/arcs/pull/3297 is migrating javaharness to pipes-shell-2.
I'll send a new PR as follow up for the equivalent of this change (with a proper description).

> Two other notes re: my other comment:
> 
> 1. I am guessing perhaps this is modifying pipes-shell because javaharness which we're using for JavaPEC work is still using pipes-shell not pipes-shell-2, can we look at moving it to use pipes-shell-2 as a separate PR before this one?
> 2. Worth adding note to the PR description on what the function need is that prompted this PR

**Pull request closed without merging**